src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -2196,13 +2196,12 @@ public static unsafe Matrix4x4 Lerp(Matrix4x4 matrix1, Matrix4x4 matrix2, float
         /// <returns>The hash code.</returns>
         public override readonly int GetHashCode()
         {
-            unchecked
-            {
-                return M11.GetHashCode() + M12.GetHashCode() + M13.GetHashCode() + M14.GetHashCode() +
-                       M21.GetHashCode() + M22.GetHashCode() + M23.GetHashCode() + M24.GetHashCode() +
-                       M31.GetHashCode() + M32.GetHashCode() + M33.GetHashCode() + M34.GetHashCode() +
-                       M41.GetHashCode() + M42.GetHashCode() + M43.GetHashCode() + M44.GetHashCode();
-            }
+            return HashCode.Combine(HashCode.Combine(
```

Comment:
For this one, rather than two combined `HashCode.Combine` calls, it would likely be better to use the:
```
var hash = new HashCode();
hash.Add(M11);
...
hash.Add(M44);
return hash.ToHashCode();
```

pattern as it avoids us finalizing halfway through. You can see some other examples where we follow this pattern here: <https://source.dot.net/#System.Private.CoreLib/HashCode.cs,47111822c49d9e91,references>

Reply:
@tannergooding  thanks for advice. What do you prefer long or short version?
nice but too long
![Screenshot_1](https://user-images.githubusercontent.com/9221694/78271361-beda6180-7514-11ea-82fe-e8ad1c011436.png)
vs short, looks better but not sure 
![Screenshot_2](https://user-images.githubusercontent.com/9221694/78271379-c3067f00-7514-11ea-8b47-2825671d8592.png)

Reply:
The official coding style documentation is here: https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md

We prefer the former even with it being longer and don't generally make exceptions for matrix like data. You might be able to do:
```csharp
var hash = new HashCode();

hash.Add(M11);
hash.Add(M12);
hash.Add(M13);
hash.Add(M14);

hash.Add(M21);
hash.Add(M22);
hash.Add(M23);
hash.Add(M24);

hash.Add(M31);
hash.Add(M32);
hash.Add(M33);
hash.Add(M34);

hash.Add(M41);
hash.Add(M42);
hash.Add(M43);
hash.Add(M44);

return hash.ToHashCode();
```

which makes it even longer, but helps break it apart into the elements of each row.

Reply:
can't use `var hash = new HashCode();` because `Do not use default value type constructor`. what about

```
           var hash1 = HashCode.Combine(M11, M12, M13, M14);
           var hash2 = HashCode.Combine(M21, M22, M23, M24);
           var hash3 = HashCode.Combine(M31, M32, M33, M34);
           var hash4 = HashCode.Combine(M41, M42, M43, M44);
            
            return HashCode.Combine(hash1, hash2, hash3, hash4);
```

or only hash1 and hash2 with 8 arguments

Reply:
> can't use var hash = new HashCode(); because Do not use default value type constructor. 

```C#
HashCode hash = default;
```

Reply:
What is reporting `Do not use default value type constructor`? That looks like a possible `StyleCop` analyzer and isn't something that we normally use or enforce for the codebase.

Reply:
> That looks like a possible StyleCop analyzer and isn't something that we normally use or enforce for the codebase.

It is, and we do :)

Reply:
We should add that to the https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md and ensure that it is an enabled analyzer that people are aware they should have installed.

Reply:
> ensure that it is an enabled analyzer that people are aware they should have installed.

It's built in to the repo. The analyzer runs on every build of every src project.

Reply:
My machine had gotten into a weird state and wasn't reporting these in VS. I seem to have resolved it now, sorry for the confusion!